cinema/serializers.py

```diff
@@ -1 +1,88 @@
-# write serializers here
+from rest_framework import serializers
+from cinema.models import (Actor,
+                           CinemaHall,
+                           Movie,
+                           MovieSession,
+                           Order,
+                           Ticket,
+                           Genre)
+
+
+class CinemaHallSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CinemaHall
+        fields = ("id", "name", "rows", "seats_in_row", "capacity")
+
+
+class GenreSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Genre
+        fields = (
+            "id",
+            "name",
+        )
+
+
+class ActorSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Actor
+        fields = ("id", "first_name", "last_name", "full_name")
+
+
+class MovieSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Movie
+        fields = ("id", "title", "duration", "description", "genres", "actors")
+
+
+class MovieListSerializer(MovieSerializer):
+    genres = serializers.SlugRelatedField(
+        many=True,
+        read_only=True,
+        slug_field="name")
+
+    actors = serializers.StringRelatedField(many=True)
+
+    class Meta:
+        model = Movie
+        fields = "__all__"
+
+
+class MovieRetrieveSerializer(MovieSerializer):
+    genres = GenreSerializer(many=True, read_only=True)
+    actors = ActorSerializer(many=True, read_only=True)
+
+
+class MovieSessionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = MovieSession
+        fields = ("id", "show_time", "movie", "cinema_hall")
+        read_only_fields = ("id",)
+
+
+class MovieSessionListSerializer(MovieSessionSerializer):
+    movie_title = serializers.CharField(source="movie.title", read_only=True)
+
+    cinema_hall_name = serializers.CharField(
+        source="cinema_hall.name",
+        read_only=True)
+
+    cinema_hall_capacity = serializers.IntegerField(
+        source="cinema_hall.capacity", read_only=True
+    )
+
+    class Meta:
+        model = MovieSession
+        fields = (
+            "id",
+            "show_time",
+            "movie_title",
+            "cinema_hall_name",
+            "cinema_hall_capacity",
+        )
+        read_only_fields = ("id",)
+
+
+class MovieSessionRetrieveSerializer(MovieSessionSerializer):
+    movie = MovieListSerializer(many=False, read_only=True)
```

Comment:
In `MovieSessionRetrieveSerializer`, the `movie` field is using `MovieListSerializer` with `many=False`. This is correct since a `MovieSession` is associated with a single `Movie`. Ensure that `MovieListSerializer` is intended to be used here, as it includes all fields of `Movie`, which might be more than needed for a retrieve view.